src/server/cdb/data/process.c

```diff
@@ -0,0 +1,493 @@
+ * of the request buffer. Once we move to later segments, the areas pointed to
+ * by these pointers will be overwritten.
+ */
+static void
+_process_set(struct response *rsp, struct request *req)
+{
+    INCR(process_metrics, set);
+    INCR(process_metrics, set_ex);
+    rsp->type = RSP_CLIENT_ERROR;
+    rsp->vstr = str2bstr(CMD_ERR_MSG);
+    log_verb("set req %p processed, rsp type %d", req, rsp->type);
+}
+
+static void
+_process_add(struct response *rsp, struct request *req)
+{
+    INCR(process_metrics, add_ex);
+    rsp->type = RSP_CLIENT_ERROR;
+    rsp->vstr = str2bstr(CMD_ERR_MSG);
+    log_verb("add req %p processed, rsp type %d", req, rsp->type);
+}
+
+static void
+_process_replace(struct response *rsp, struct request *req)
+{
+    INCR(process_metrics, replace_ex);
+    rsp->type = RSP_CLIENT_ERROR;
+    rsp->vstr = str2bstr(CMD_ERR_MSG);
+    log_verb("replace req %p processed, rsp type %d", req, rsp->type);
+}
+
+static void
+_process_cas(struct response *rsp, struct request *req)
+{
+    INCR(process_metrics, cas_ex);
+    rsp->type = RSP_CLIENT_ERROR;
+    rsp->vstr = str2bstr(CMD_ERR_MSG);
+    log_verb("cas req %p processed, rsp type %d", req, rsp->type);
+}
+
+static void
+_process_incr(struct response *rsp, struct request *req)
+{
+    INCR(process_metrics, incr);
+    INCR(process_metrics, incr_ex);
+    rsp->type = RSP_CLIENT_ERROR;
+    rsp->vstr = str2bstr(CMD_ERR_MSG);
+    log_verb("incr req %p processed, rsp type %d", req, rsp->type);
+}
+
+static void
+_process_decr(struct response *rsp, struct request *req)
+{
+    INCR(process_metrics, decr);
+    INCR(process_metrics, decr_ex);
+    rsp->type = RSP_CLIENT_ERROR;
+    rsp->vstr = str2bstr(CMD_ERR_MSG);
+    log_verb("decr req %p processed, rsp type %d", req, rsp->type);
+}
+
+static void
+_process_append(struct response *rsp, struct request *req)
+{
+    INCR(process_metrics, append_ex);
+    rsp->type = RSP_CLIENT_ERROR;
+    rsp->vstr = str2bstr(CMD_ERR_MSG);
+    log_verb("append req %p processed, rsp type %d", req, rsp->type);
+}
+
+static void
+_process_prepend(struct response *rsp, struct request *req)
+{
+    INCR(process_metrics, prepend_ex);
+    rsp->type = RSP_CLIENT_ERROR;
+    rsp->vstr = str2bstr(CMD_ERR_MSG);
+    log_verb("prepend req %p processed, rsp type %d", req, rsp->type);
+}
+
+static void
+_process_flush(struct response *rsp, struct request *req)
+{
+    INCR(process_metrics, flush);
+    rsp->type = RSP_CLIENT_ERROR;
+    rsp->vstr = str2bstr(CMD_ERR_MSG);
+    log_info("flush req %p processed, rsp type %d", req, rsp->type);
+}
+
+void
+process_request(struct response *rsp, struct request *req)
+{
+    log_verb("processing req %p, write rsp to %p", req, rsp);
+    INCR(process_metrics, process_req);
+
+    switch (req->type) {
+    case REQ_GET:
+        _process_get(rsp, req);
+        break;
+
+    case REQ_GETS:
+        _process_gets(rsp, req);
+        break;
+
+    case REQ_DELETE:
```

Comment:
I think you may save some space by collapsing these cases and apply something like `_process_unsupported()` to all of them.